src/web/crate_details.rs

```diff
@@ -32,6 +32,8 @@ use std::sync::Arc;
 
 // TODO: Add target name and versions
 
+#[derive(Template)]
+#[template(path = "crate/details.html")]
```

Comment:
Is there a reason why in particular? `CrateDetailsPage` is exactly the same type and use most of the fields from `CrateDetails`, so not sure having this separation makes much sense. If you look at each commit, I started by moving one field after another directly into `CrateDetails` but at some point, I just merged both types because it didn't make sense.

Reply:
having thought about it, it's still feels consusing for users of that struct that it carries a `csp_nonce` around, which is only used in one view.

Other (current) users of that struct (`target_redirect_handler`, `rustdoc_html_server_handler`, `rustdoc_redirector_handler`) would also get it and it would be empty?